wqflask/wqflask/correlation_matrix/show_corr_matrix.py

```diff
@@ -187,66 +186,29 @@ def __init__(self, start_vars):
                             samples=self.all_sample_list,
                             sample_data=self.sample_data,)
 
-    def calculate_pca(self, cols, corr_eigen_value, corr_eigen_vectors):
-        base = importr('base')
-        stats = importr('stats')
+    def calculate_pca(self):
 
-        corr_results_to_list = ro.FloatVector(
-            [item for sublist in self.pca_corr_results for item in sublist])
+        pca = compute_pca(self.pca_corr_results)
 
-        m = ro.r.matrix(corr_results_to_list, nrow=len(cols))
-        eigen = base.eigen(m)
-        pca = stats.princomp(m, cor="TRUE")
-        self.loadings = pca.rx('loadings')
-        self.scores = pca.rx('scores')
-        self.scale = pca.rx('scale')
-
-        trait_array = zScore(self.trait_data_array)
-        trait_array_vectors = np.dot(corr_eigen_vectors, trait_array)
-
-        pca_traits = []
-        for i, vector in enumerate(trait_array_vectors):
-            # ZS: Check if below check is necessary
-            # if corr_eigen_value[i-1] > 100.0/len(self.trait_list):
-            pca_traits.append((vector * -1.0).tolist())
+        self.loadings = pca["components"]
+        self.scores = pca["scores"]
 
         this_group_name = self.trait_list[0][1].group.name
         temp_dataset = data_set.create_dataset(
             dataset_name="Temp", dataset_type="Temp", group_name=this_group_name)
         temp_dataset.group.get_samplelist()
-        for i, pca_trait in enumerate(pca_traits):
-            trait_id = "PCA" + str(i + 1) + "_" + temp_dataset.group.species + "_" + \
-                this_group_name + "_" + datetime.datetime.now().strftime("%m%d%H%M%S")
-            this_vals_string = ""
-            position = 0
-            for sample in temp_dataset.group.all_samples_ordered():
-                if sample in self.shared_samples_list:
-                    this_vals_string += str(pca_trait[position])
-                    this_vals_string += " "
-                    position += 1
-                else:
-                    this_vals_string += "x "
-            this_vals_string = this_vals_string[:-1]
 
-            Redis.set(trait_id, this_vals_string, ex=THIRTY_DAYS)
-            self.pca_trait_ids.append(trait_id)
+        pca_dataset = generate_pca_temp_dataset(species=temp_dataset.group.species, group=this_group_name,
+                                                traits_data=self.trait_data_array, corr_array=self.pca_corr_results,
+                                                dataset_samples=temp_dataset.group.all_samples_ordered(),
+                                                shared_samples=self.shared_samples_list,
+                                                create_time=datetime.datetime.now().strftime("%m%d%H%M%S"))
 
-        return pca
+        cache_pca_dataset(Redis, THIRTY_DAYS, pca_dataset)
```

Comment:
**issue**: @zsloan do we need to make this `THIRTY_DAYS` configurable. Also @Alexanderlacuna it'd be nice to make the end caller of this function responsible for setting this.

Reply:
Hm, probably. It might be good to just have a generic configuration for how long temp traits (whether from PCA or from something else) are stored in Redis.